sycl/include/sycl/esimd/esimd_enum.hpp

```diff
@@ -0,0 +1,137 @@
+// Mark a function being nodebug.
+#define ESIMD_NODEBUG __attribute__((nodebug))
+// Mark a "ESIMD global": accessible from all functions in current translation
+// unit, separate copy per subgroup (work-item), mapped to SPIRV private storage
+// class.
+#define ESIMD_PRIVATE __attribute__((opencl_private))
+// Bind a ESIMD global variable to a specific register.
+#define ESIMD_REGISTER(n) __attribute__((register(n)))
+#else
+// TODO ESIMD define what this means on Windows host
+#define ESIMD_NODEBUG
+// On host device ESIMD global is a thread local static var. This assumes that
+// each work-item is mapped to a separate OS thread on host device.
+#define ESIMD_PRIVATE thread_local
+#define ESIMD_REGISTER(n)
+#endif
+
+// Mark a function being noinline
+#define ESIMD_NOINLINE __attribute__((noinline))
+// Mark a function to be inlined
+#define ESIMD_INLINE __attribute__((always_inline))
+
+// Enums
+enum { GENX_NOSAT = 0, GENX_SAT };
+
+enum ChannelMaskType {
+  ESIMD_R_ENABLE = 1,
+  ESIMD_G_ENABLE = 2,
+  ESIMD_GR_ENABLE = 3,
+  ESIMD_B_ENABLE = 4,
+  ESIMD_BR_ENABLE = 5,
+  ESIMD_BG_ENABLE = 6,
+  ESIMD_BGR_ENABLE = 7,
+  ESIMD_A_ENABLE = 8,
+  ESIMD_AR_ENABLE = 9,
+  ESIMD_AG_ENABLE = 10,
+  ESIMD_AGR_ENABLE = 11,
+  ESIMD_AB_ENABLE = 12,
+  ESIMD_ABR_ENABLE = 13,
+  ESIMD_ABG_ENABLE = 14,
+  ESIMD_ABGR_ENABLE = 15
+};
+
+#define NumChannels(Mask)                                                      \
+  ((Mask & 1) + ((Mask & 2) >> 1) + ((Mask & 4) >> 2) + ((Mask & 8) >> 3))
+
+#define HasR(Mask) ((Mask & 1) == 1)
+
+#define HasG(Mask) ((Mask & 2) >> 1 == 1)
+
+#define HasB(Mask) ((Mask & 4) >> 2 == 1)
+
+#define HasA(Mask) ((Mask & 8) >> 3 == 1)
+
+enum class EsimdAtomicOpType {
+  ATOMIC_ADD = 0x0,
+  ATOMIC_SUB = 0x1,
+  ATOMIC_INC = 0x2,
+  ATOMIC_DEC = 0x3,
+  ATOMIC_MIN = 0x4,
+  ATOMIC_MAX = 0x5,
+  ATOMIC_XCHG = 0x6,
+  ATOMIC_CMPXCHG = 0x7,
+  ATOMIC_AND = 0x8,
+  ATOMIC_OR = 0x9,
+  ATOMIC_XOR = 0xa,
+  ATOMIC_MINSINT = 0xb,
+  ATOMIC_MAXSINT = 0xc,
+  ATOMIC_FMAX = 0x10,
+  ATOMIC_FMIN = 0x11,
+  ATOMIC_FCMPWR = 0x12,
+  ATOMIC_PREDEC = 0xff
+};
+
+#define ATOMIC_ADD EsimdAtomicOpType::ATOMIC_ADD
+#define ATOMIC_SUB EsimdAtomicOpType::ATOMIC_SUB
+#define ATOMIC_INC EsimdAtomicOpType::ATOMIC_INC
+#define ATOMIC_DEC EsimdAtomicOpType::ATOMIC_DEC
+#define ATOMIC_MIN EsimdAtomicOpType::ATOMIC_MIN
+#define ATOMIC_MAX EsimdAtomicOpType::ATOMIC_MAX
+#define ATOMIC_XCHG EsimdAtomicOpType::ATOMIC_XCHG
+#define ATOMIC_CMPXCHG EsimdAtomicOpType::ATOMIC_CMPXCHG
+#define ATOMIC_AND EsimdAtomicOpType::ATOMIC_AND
+#define ATOMIC_OR EsimdAtomicOpType::ATOMIC_OR
+#define ATOMIC_XOR EsimdAtomicOpType::ATOMIC_XOR
+#define ATOMIC_MINSINT EsimdAtomicOpType::ATOMIC_MINSINT
+#define ATOMIC_MAXSINT EsimdAtomicOpType::ATOMIC_MAXSINT
+#define ATOMIC_FMAX EsimdAtomicOpType::ATOMIC_FMAX
+#define ATOMIC_FMIN EsimdAtomicOpType::ATOMIC_FMIN
+#define ATOMIC_FCMPWR EsimdAtomicOpType::ATOMIC_FCMPWR
+#define ATOMIC_PREDEC EsimdAtomicOpType::ATOMIC_PREDEC
```

Comment:
Still a lot of macros that might conflict...

Reply:
fixed.